devito/stencilkernel.py

```diff
@@ -112,14 +116,25 @@ def __init__(self, stencils, **kwargs):
 
         # Now resolve and substitute dimensions for loop index variables
         subs = {}
-        nodes = ResolveIterationVariable().visit(nodes, subs=subs)
+        variable_map = defaultdict(int)
```

Comment:
I tried that, but I got some bizarre pseudo-caching that caused breakages and then I ran out of patience. I'll try again and investigate though.